tests/E2E/SnapshotTest.php

```diff
@@ -56,4 +57,45 @@ public function testGetEntrySnapshots()
         $this->assertInstanceOf(EntrySnapshot::class, $snapshots[0]);
         $this->assertCount(1, $snapshots);
     }
+
+    /**
+     * @vcr e2e_snapshot_content_type_get.json
+     */
+    public function testGetContentTypeSnapshot()
+    {
+        $manager = $this->getReadWriteSpaceManager();
+
+        $snapshot = $manager->getContentTypeSnapshot('versionedContentType', '1Qvx64r3Nq0MOtftdcAXJO');
+
+        $contentType = $snapshot->getContentType();
+        $this->assertEquals('Versioned Content Type', $contentType->getName());
+        $this->assertEquals('title', $contentType->getDisplayField());
+        $this->assertEquals(2, $contentType->getSystemProperties()->getVersion());
+        $this->assertEquals(1, $contentType->getSystemProperties()->getPublishedCounter());
+
+        $sys = $snapshot->getSystemProperties();
+        $this->assertEquals('ContentType', $sys->getSnapshotEntityType());
+        $this->assertEquals('publish', $sys->getSnapshotType());
+        $this->assertEquals(new Link($this->readWriteSpaceId, 'Space'), $sys->getSpace());
+        $this->assertEquals(new \DateTimeImmutable('2017-07-05T16:46:58.486Z'), $sys->getCreatedAt());
+        $this->assertEquals(new \DateTimeImmutable('2017-07-05T16:46:58.486Z'), $sys->getUpdatedAt());
+    }
+
+    /**
+     * @vcr e2e_snapshot_content_type_get_collection.json
+     */
+    public function testGetContentTypeSnapshots()
+    {
+        $manager = $this->getReadWriteSpaceManager();
+
+        $snapshots = $manager->getContentTypeSnapshots('versionedContentType');
+
+        $this->assertInstanceOf(ContentTypeSnapshot::class, $snapshots[0]);
+
+        $query = (new Query())
+            ->setLimit(1);
+        $snapshots = $manager->getContentTypeSnapshots('versionedContentType', $query);
+        $this->assertInstanceOf(ContentTypeSnapshot::class, $snapshots[0]);
+        $this->assertCount(1, $snapshots);
```

Comment:
I could, though I don't feel it'd be necessary.
The builder of the collection endpoint under the hood uses the builder of the single endpoint, which means that the fact that this gets executed without errors is already a guarantee that the `ContentTypeSnapshot` objects are correctly built.
Not sure if I was able to explain myself 😅